Source/WebCore/css/CSSGradientValue.h

```diff
@@ -232,38 +238,48 @@ class CSSRadialGradientValue final : public CSSValue {
     struct Shape {
         ShapeKeyword shape;
         std::optional<CSSGradientPosition> position;
+        friend bool operator==(const Shape&, const Shape&) = default;
     };
     struct Extent {
         ExtentKeyword extent;
         std::optional<CSSGradientPosition> position;
+        friend bool operator==(const Extent&, const Extent&) = default;
     };
     struct Length {
         Ref<CSSPrimitiveValue> length; // <length [0,∞]>
         std::optional<CSSGradientPosition> position;
+
+        friend bool operator==(const Length& a, const Length&);
     };
     struct CircleOfLength {
         Ref<CSSPrimitiveValue> length; // <length [0,∞]>
         std::optional<CSSGradientPosition> position;
+        friend bool operator==(const CircleOfLength&, const CircleOfLength&);
     };
     struct CircleOfExtent {
         ExtentKeyword extent;
         std::optional<CSSGradientPosition> position;
+        friend bool operator==(const CircleOfExtent&, const CircleOfExtent&) = default;
     };
     struct Size {
         std::pair<Ref<CSSPrimitiveValue>, Ref<CSSPrimitiveValue>> size; // <length-percentage [0,∞]>, <length-percentage [0,∞]>
         std::optional<CSSGradientPosition> position;
+        friend bool operator==(const Size& a, const Size&);
     };
     struct EllipseOfSize {
         std::pair<Ref<CSSPrimitiveValue>, Ref<CSSPrimitiveValue>> size; // <length-percentage [0,∞]>, <length-percentage [0,∞]>
         std::optional<CSSGradientPosition> position;
+        friend bool operator==(const EllipseOfSize&, const EllipseOfSize&);
```

Comment:
This is missing the `= default`.

Reply:
This is not defaulted, it has an implementation in the cpp. However, I add to add the declaration in the header otherwise some of the defaulted comparison operators were complaining.

The same comment applies to the other places you commented on.